src/metabase/upload.clj

```diff
@@ -513,6 +514,38 @@
            (field->db-type driver field->new-type)
            args)))
 
+(defn- mbql? [model]
+  (= "query" (name (:query_type model "query"))))
+
+(defn- no-joins?
+  "Returns true if `query` has no joins in it, otherwise false."
```

Comment:
Not your fault, but I realised this function (written by me) has a small logical bug in that it returns true even if the query has implicit joins in it. We should be explicit here in the docstring:
```suggestion
  "Returns true if `query` has no explicit joins in it, otherwise false."
```

Reply:
Model FKs are currently not implemented, and CSV upload tables are unlikely to have FKs set up, because that requires database privileges most CSV upload users wouldn't use. But that might change so you could leave a TODO here for now